File: reverse_integer.py
def reverse_integer(x):
    """
    Reverse digits of an integer.
    
    :param x: Integer to be reversed
    :return: Reversed integer
    """
    sign = -1 if x < 0 else 1  # Determine the sign of the integer
    x *= sign  # Make the integer positive for reversal
    reversed_x = 0  # Initialize the reversed integer

    while x:
        reversed_x = reversed_x * 10 + x % 10  # Add the last digit of x to reversed_x
        x //= 10  # Remove the last digit from x

    reversed_x *= sign  # Restore the original sign

    # Check if the reversed integer is within the 32-bit signed integer range
    if reversed_x < -2**31 or reversed_x > 2**31 - 1:
        return 0  # Return 0 if it is out of range
    return reversed_x

# Example usage
x = 123
print(reverse_integer(x))  # Output should be 321

File: reverse_integer.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main() {
    // Declare a double variable 'r' to store the value 2^31
    double r = pow(2, 31);

    // Declare integer variables 'number' and 'digit'
    int number;
    int digit;

    // Prompt the user to enter a number
    cout << "Please enter a number: ";
    cin >> number;

    // Check if the number is within the range of -2^31 to 2^31
    if (number <= r && number >= -r) {
        // If the number is negative, print the minus sign and make the number positive
        if (number < 0) {
            number *= -1;
            cout << "-";
        }

        // Loop to extract and print each digit of the number in reverse order
        while (number > 0) {
            digit = number % 10;  // Get the last digit of the number
            number = number / 10; // Remove the last digit from the number
            cout << digit;        // Print the digit
        }
    }

    return 0;
}
